src/libs/auth.ts

```diff
@@ -109,59 +149,146 @@ const getSigninArgs = (includeBillingScope) => {
+        ...state,
+        refreshTokenMetadata: {
+          id: uuid(),
+          token: refreshToken,
+          createdAt: authTokenCreatedAt,
+          // I do not think we have any way of getting this info except by looking at B2C configs
+          expiresAt: authTokenCreatedAt + 86400, // 24 hours in seconds
+        },
+      }));
+    }
+    authStore.update((state) => ({
+      ...state,
+      authTokenMetadata: {
+        ...authStore.get().authTokenMetadata,
+        createdAt: authTokenCreatedAt,
+        expiresAt: authTokenExpiresAt,
+        id: authTokenId,
+        totalAuthTokensUsedThisSession: authStore.get().authTokenMetadata.totalAuthTokensUsedThisSession + 1,
+      },
+    }));
+    Ajax().Metrics.captureEvent(Events.user.authTokenLoad.success, {
+      authProvider: user.profile.idp,
+      ...labelOldAuthToken(oldAuthTokenMetadata),
+      authTokenCreatedAt: labelTimestampMetric(authTokenCreatedAt),
+      authTokenExpiresAt: labelTimestampMetric(authTokenExpiresAt),
+      authTokenId,
+      refreshTokenId: authStore.get().refreshTokenMetadata.id,
+      refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+      refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+      jwtExpiresAt: labelTimestampMetric(jwtExpiresAt),
+    });
+  } else if (reloadedAuthTokenState === null) {
+    Ajax().Metrics.captureEvent(Events.user.authTokenLoad.expired, {
+      ...labelOldAuthToken(oldAuthTokenMetadata),
+      refreshTokenId: authStore.get().refreshTokenMetadata.id,
+      refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+      refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+    });
+  } else {
+    Ajax().Metrics.captureEvent(Events.user.authTokenLoad.error, {
+      // we could potentially log the reason, but I don't know if that data is safe to log
+      ...labelOldAuthToken(oldAuthTokenMetadata),
+      refreshTokenId: authStore.get().refreshTokenMetadata.id,
+      refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+      refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+    });
+  }
+  return reloadedAuthTokenState;
+};
+const labelOldAuthToken = (oldAuthTokenMetadata) => {
+  return {
+    oldAuthTokenCreatedAt: labelTimestampMetric(oldAuthTokenMetadata.createdAt),
+    oldAuthTokenExpiresAt: labelTimestampMetric(oldAuthTokenMetadata.expiresAt),
+    oldAuthTokenId: oldAuthTokenMetadata.id === undefined ? 'undefined' : oldAuthTokenMetadata.id,
+    oldAuthTokenLifespan:
+      oldAuthTokenMetadata.id === undefined ? 'undefined' : Date.now() - oldAuthTokenMetadata.createdAt / 1000.0,
+  };
+};
+const labelTimestampMetric = (timestamp: number): string => {
+  return timestamp < 0 ? 'timestamp unset' : Utils.formatTimestampInSeconds(timestamp);
```

Comment:
"undefined" is used for other event fields. Why is this one different?

```suggestion
  return timestamp < 0 ? 'undefined' : Utils.formatTimestampInSeconds(timestamp);
```

Reply:
We wanted to make this clear that this value had not been set so it did not look like a bug in mixpanel. In this case, a more specific message seemed helpful. If it is more confusing than helpful, we can definitely pivot.